packages/core/src/convertConfig.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  AggregateBy, Bookmark, BookmarkedSelection, Column, ColumnName, Histogram, PlotInformation, Row, Scatterplot,
+  SortByOrder, SortVisibleBy, UpsetConfig,
+} from './types';
+import { isUpsetConfig } from './typecheck';
+
+/**
+ * Developer notes:
+ * This file is used to convert old versions of the config to the current version.
+ * This conversion avoids breaking changes when the config format changes.
+ * To add a new version:
+ * 1. Copy the old UpsetConfig type into this file and give it a new, unique name.
+ * 2. Make your changes to UpsetConfig in types.ts
+ * 3. Change the return type of the most recent version conversion function to the old type that you copied.
+ * 4. Implement a version conversion function that takes the old type and returns the new type (UpsetConfig).
+ *    This function must modify the input config in place.
+ * 5. Add a new case to the switch statement in convertConfig.
+ *    - The case version number should be the current version number, as your conversion function starts
+ *      with the previous version (which the current version will become) and converts it to the current version.
+ *    - The only statement within the case body should run your conversion function on the `config` var.
+ *    - Do not use break statements in the switch, as conversions should be applied cumulatively & in order.
+ * 6. Update isUpsetConfig (in typecheck.ts) to reflect your changes to UpsetConfig.
+ * 7. Bump the version number in the UpsetConfig type, all package.json files, the README, and defaultConfig.ts.
+ */
```

Comment:
Should this also be added to the developer documentation on the site? Where do I edit that?

Reply:
Documentation for this repo is automatically generated using docstrings, but the README.md file for the repo also contains a section (very bottom) that is for development guidelines. Right now I think there is only a documentation section, but this should for sure be added. I'd suggest making a new header for Developer Guidelines and then have documentation as a sub heading. The doc guidelines already exist so no need to add or remove. This could go into a "notes" subheading or similar.

Reply:
Section added in 6a9a253610edbec6717e14482e1e48412956b6a8